ParkingLot/com/parkinglot/ParkingLotService.java

```diff
@@ -0,0 +1,147 @@
+package com.parkinglot;
+
+import com.parkinglot.models.Floor;
+import com.parkinglot.models.ParkingLot;
+import com.parkinglot.models.Slot;
+import com.parkinglot.models.Vehicle;
+import com.parkinglot.models.VehicleType;
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
+public class ParkingLotService {
+  private Map<Floor, List<Slot>> floorSlotMap = new LinkedHashMap<>();
+  private ParkingLot parkingLot = null;
+  public void createParkingLot(String parkingId,  int numberOfFloors, int numberOfSlots ) {
+    parkingLot = new ParkingLot(parkingId);
+    for (int floorNumber = 0; floorNumber < numberOfFloors; floorNumber++) {
+      List<Slot> slots = new ArrayList<>();
+      for (int slotNumber = 0; slotNumber < numberOfSlots; slotNumber++) {
+        VehicleType vehicleType;
+        Slot slot;
+        if (slotNumber == 0) {
+          vehicleType = VehicleType.TRUCK;
+          slot = new Slot(slotNumber, vehicleType);
+        } else if (slotNumber == 1 || slotNumber == 2) {
+          vehicleType = VehicleType.BIKE;
+          slot = new Slot(slotNumber, vehicleType);
+        } else {
+          vehicleType = VehicleType.CAR;
+          slot = new Slot(slotNumber, vehicleType);
+        }
+        slots.add(slot);
+      }
+      Floor floor = new Floor(floorNumber);
+      floorSlotMap.put(floor,slots);
+    }
+    System.out.println("Created a parking lot with " + numberOfFloors + " floors and " + numberOfSlots + " per floor");
+
+  }
+
+  public void displaySlots(String displayType, String vehicleTypeToCheck) {
+    switch (displayType) {
+      case "free_count" -> displayAvailableSlotsCount(vehicleTypeToCheck);
+      case "free_slots" -> displayAvailableSlots(vehicleTypeToCheck);
+      case "occupied_slots" -> displayOccupiedSlots(vehicleTypeToCheck);
+    }
+  }
+
+  private void displayOccupiedSlots(String vehicleTypeToCheck) {
+    int occupiedSlotNum;
+    for (Map.Entry<Floor, List<Slot>> allFloors : floorSlotMap.entrySet()) {
+      StringBuilder occupiedSlots = new StringBuilder();
+      int floorNumberToDisplay = allFloors.getKey().getFloorNumber()+ 1;
+      for (Slot slotNumber : allFloors.getValue()) {
+        if (slotNumber.getVehicle() != null && slotNumber.getVehicleType().toString()
```

Comment:
DRY

Reply:
What is repetitive here?

Reply:
I can see the same conditions in other functions. What if you've to change the logic? Do you want to change it everywhere or at only one place?